modules/thermal_hydraulics/src/components/ShaftConnectedTurbine1Phase.C

```diff
@@ -145,22 +145,21 @@ ShaftConnectedTurbine1Phase::addVariables()
   VolumeJunction1Phase::addVariables();
 
   getTHMProblem().addSimVariable(false, _delta_p_var_name, FEType(FIRST, SCALAR));
-  getTHMProblem().addConstantScalarIC(_delta_p_var_name, 0);
-
   getTHMProblem().addSimVariable(false, _power_var_name, FEType(FIRST, SCALAR));
-  getTHMProblem().addConstantScalarIC(_power_var_name, 0);
-
   getTHMProblem().addSimVariable(false, _driving_torque_var_name, FEType(FIRST, SCALAR));
-  getTHMProblem().addConstantScalarIC(_driving_torque_var_name, 0);
-
   getTHMProblem().addSimVariable(false, _flow_coeff_var_name, FEType(FIRST, SCALAR));
-  getTHMProblem().addConstantScalarIC(_flow_coeff_var_name, 0);
-
   getTHMProblem().addSimVariable(false, _friction_torque_var_name, FEType(FIRST, SCALAR));
-  getTHMProblem().addConstantScalarIC(_friction_torque_var_name, 0);
-
   getTHMProblem().addSimVariable(false, _moment_of_inertia_var_name, FEType(FIRST, SCALAR));
-  getTHMProblem().addConstantScalarIC(_moment_of_inertia_var_name, _inertia_const);
+
+  if (!_app.isRestarting())
+  {
+    getTHMProblem().addConstantScalarIC(_delta_p_var_name, 0);
+    getTHMProblem().addConstantScalarIC(_power_var_name, 0);
+    getTHMProblem().addConstantScalarIC(_driving_torque_var_name, 0);
+    getTHMProblem().addConstantScalarIC(_flow_coeff_var_name, 0);
+    getTHMProblem().addConstantScalarIC(_friction_torque_var_name, 0);
+    getTHMProblem().addConstantScalarIC(_moment_of_inertia_var_name, _inertia_const);
```

Comment:
I still think this kind of think is problematic.
If the user passes a new inertia_const when doing their restart, it gets ignored

Reply:
ah that works